src/components/lib/cos_kernel_api.c

```diff
@@ -298,7 +299,7 @@ __page_bump_valloc(struct cos_compinfo *ci)
 
 		/* PTE */
 		if (call_cap_op(meta->captbl_cap, CAPTBL_OP_PGTBLACTIVATE,
-				pte_cap, meta->pgtbl_cap, ptemem_cap, 1)) {
+				pte_cap, meta->local_mi.pgtbl_cap, ptemem_cap, 1)) {
```

Comment:
See above comment. In this case, pte_cap has to be a cap in **our own** cap table.
We provide `meta->captbl_cap` because we modify **our own** cap table.
The interesting one is `ptemem_cap`. It is a cap in page table instead of cap table.
So we have to provide a cap to that page table, which is `meta->local_mi.pgtbl_cap`
Of course we need to be careful if you provide the correct page table